src/rabbit_mgmt_wm_definitions.erl

```diff
@@ -426,3 +437,71 @@ rv(VHost, Type, Props) -> rv(VHost, Type, name, Props).
 
 rv(VHost, Type, Name, Props) ->
     rabbit_misc:r(VHost, Type, maps:get(Name, Props, undefined)).
+
+%%--------------------------------------------------------------------
+
+validate_limits(All) ->
+    case maps:get(queues, All, undefined) of
+        undefined -> ok;
+        Queues0 ->
+            {ok, VHostMap} = filter_out_existing_queues(Queues0),
+            maps:fold(fun validate_vhost_limit/3, ok, VHostMap)
+    end.
+
+validate_limits(All, VHost) ->
+    case maps:get(queues, All, undefined) of
+        undefined -> ok;
+        Queues0 ->
+            Queues1 = filter_out_existing_queues(VHost, Queues0),
+            AddCount = length(Queues1),
+            validate_vhost_limit(VHost, AddCount, ok)
+    end.
+
+filter_out_existing_queues(Queues) ->
+    QData = [ build_queue_data(Q) || Q <- Queues ],
+    build_filtered_map(QData, maps:new()).
+
+build_queue_data(Q) ->
+    VHost = maps:get(<<"vhost">>, Q, undefined),
+    Rec = rv(VHost, queue, <<"name">>, Q),
+    {Rec, VHost}.
+
+build_filtered_map([], AccMap) ->
+    {ok, AccMap};
+build_filtered_map([{Rec, VHost}|Rest], AccMap0) ->
+    case rabbit_amqqueue:lookup(Rec) of
+        {error, not_found} ->
+            Count0 = maps:get(VHost, AccMap0, 0),
```

Comment:
This update could be simplified with: `AccMap1 = maps:update_with(VHost, fun(V) -> V + 1 end, AccMap0)`